src/rustdoc.md

```diff
@@ -36,30 +36,141 @@ does is call the `main()` that's in this crate's `lib.rs`, though.)
 
 ## Cheat sheet
 
+* Run `./x.py setup tools` before getting started. This will configure `x.py`
+  with nice settings for developing rustdoc and other tools, including
+  downloading a copy of rustc rather than building it.
 * Use `./x.py build` to make a usable
   rustdoc you can run on other projects.
   * Add `library/test` to be able to use `rustdoc --test`.
-  * If you've used `rustup toolchain link local /path/to/build/$TARGET/stage1`
-    previously, then after the previous build command, `cargo +local doc` will
-    Just Work.
+  * Run `rustup toolchain link local /path/to/build/$TARGET/stage1` to add a
+    custom toolchain called `local` to your rustup environment. After running
+    that, `cargo +local doc` in any directory will build with your
+    locally-compiled rustdoc.
```

Comment:
FWIW, this doesn't work correctly with download-rustc on. Instead, you need to use stage2.

We don't have to address that here, but perhaps change the toolchain name from `local` to `stage1` so it's easier to tell the two stages apart?

Reply:
Yeah, I think this should be using the stage2 directory as well since it recommended download-rustc.

Reply:
Fixed, thanks!